moveit_core/collision_detection/include/moveit/collision_detection/world.h

```diff
@@ -44,12 +44,15 @@
 #include <map>
 #include <memory>
 #include <boost/function.hpp>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
Just to make sure I understand the logic here - we don't want to use std::shared_ptr for ``Shape`` because its part of geometric_shapes, which is part of ``desktop-full``, which is not suppose to have std in its public headers yet, correct?

https://github.com/ros-planning/geometric_shapes/blob/kinetic-devel/include/geometric_shapes/shapes.h#L273